Postgres JDBC 42.0.0 produces tons of error logs on container startup

```
2017-03-24 18:56:50.647 ERROR --- [      pool-3-thread-2] o.p.Driver                               : Connection error: 
org.postgresql.util.PSQLException: The connection attempt failed.
	at org.postgresql.core.v3.ConnectionFactoryImpl.openConnectionImpl(ConnectionFactoryImpl.java:275) ~[postgresql-42.0.0.jar:42.0.0]
	at org.postgresql.core.ConnectionFactory.openConnection(ConnectionFactory.java:49) ~[postgresql-42.0.0.jar:42.0.0]
	at org.postgresql.jdbc.PgConnection.<init>(PgConnection.java:194) ~[postgresql-42.0.0.jar:42.0.0]
	at org.postgresql.Driver.makeConnection(Driver.java:431) ~[postgresql-42.0.0.jar:42.0.0]
	at org.postgresql.Driver.connect(Driver.java:247) ~[postgresql-42.0.0.jar:42.0.0]
	at org.testcontainers.containers.JdbcDatabaseContainer.lambda$createConnection$2(JdbcDatabaseContainer.java:123) ~[jdbc-1.2.0.jar:?]
	at org.rnorth.ducttape.unreliables.Unreliables.lambda$retryUntilSuccess$0(Unreliables.java:41) ~[duct-tape-1.0.6.jar:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_121]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [?:1.8.0_121]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [?:1.8.0_121]
	at java.lang.Thread.run(Thread.java:745) [?:1.8.0_121]
Caused by: java.io.EOFException
	at org.postgresql.core.PGStream.receiveChar(PGStream.java:284) ~[postgresql-42.0.0.jar:42.0.0]
	at org.postgresql.core.v3.ConnectionFactoryImpl.doAuthentication(ConnectionFactoryImpl.java:418) ~[postgresql-42.0.0.jar:42.0.0]
	at org.postgresql.core.v3.ConnectionFactoryImpl.openConnectionImpl(ConnectionFactoryImpl.java:222) ~[postgresql-42.0.0.jar:42.0.0]
	... 10 more
```
With `9.4.1212` they are produced as well, but not logged.

Probably this change made them appear by default with some logging configurations:
https://www.postgresql.org/about/news/1736/
> java.util.logging is now used for logging: logging documentation

Is it something that could be fixed on your side or I should just disable all logging for PG driver?
Feels, like a bad idea...

--- a/core/src/main/java/org/testcontainers/containers/output/OutputFrame.java
+++ b/core/src/main/java/org/testcontainers/containers/output/OutputFrame.java
@@ -30,7 +30,7 @@ public class OutputFrame {
     public String getUtf8String() {
 
         if (bytes == null) {
-            return null;
+            return "";
         }
 
         return new String(bytes, Charsets.UTF_8);

--- a/modules/jdbc-test/pom.xml
+++ b/modules/jdbc-test/pom.xml
@@ -34,7 +34,7 @@
         <dependency>
             <groupId>org.postgresql</groupId>
             <artifactId>postgresql</artifactId>
-            <version>9.3-1101-jdbc41</version>
+            <version>42.0.0</version>
             <scope>test</scope>
         </dependency>
         <dependency>

--- a/modules/postgresql/src/main/java/org/testcontainers/containers/PostgreSQLContainer.java
+++ b/modules/postgresql/src/main/java/org/testcontainers/containers/PostgreSQLContainer.java
@@ -1,5 +1,12 @@
 package org.testcontainers.containers;
 
+import org.testcontainers.containers.output.OutputFrame;
+import org.testcontainers.containers.output.WaitingConsumer;
+
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.function.Predicate;
+
 /**
  * @author richardnorth
  */
@@ -72,5 +79,24 @@ public class PostgreSQLContainer<SELF extends PostgreSQLContainer<SELF>> extends
     public SELF withPassword(final String password) {
         this.password = password;
         return self();
+    }
+
+    @Override
+    protected void waitUntilContainerStarted() {
+        String regEx = ".*database system is ready to accept connections.*\\s";
+        logger().info("Waiting for database to log '{}'", regEx);
+
+        WaitingConsumer waitingConsumer = new WaitingConsumer();
+        this.followOutput(waitingConsumer);
+
+
+        Predicate<OutputFrame> waitPredicate = outputFrame ->
+                outputFrame.getUtf8String().matches(regEx);
+
+        try {
+            waitingConsumer.waitUntil(waitPredicate, 120, TimeUnit.SECONDS, 2);
+        } catch (TimeoutException e) {
+            throw new ContainerLaunchException("Timed out waiting for log output matching '" + regEx + "'");
+        }
     }
 }

--- a/modules/postgresql/src/main/java/org/testcontainers/containers/PostgreSQLContainer.java
+++ b/modules/postgresql/src/main/java/org/testcontainers/containers/PostgreSQLContainer.java
@@ -1,18 +1,17 @@
 package org.testcontainers.containers;
 
-import org.testcontainers.containers.output.OutputFrame;
-import org.testcontainers.containers.output.WaitingConsumer;
+import org.testcontainers.containers.wait.LogMessageWaitStrategy;
 
-import java.util.concurrent.TimeUnit;
-import java.util.concurrent.TimeoutException;
-import java.util.function.Predicate;
+import java.time.Duration;
+
+import static java.time.temporal.ChronoUnit.SECONDS;
 
 /**
  * @author richardnorth
  */
 public class PostgreSQLContainer<SELF extends PostgreSQLContainer<SELF>> extends JdbcDatabaseContainer<SELF> {
-    public static final String NAME = "postgresql";
-    public static final String IMAGE = "postgres";
+    static final String NAME = "postgresql";
+    static final String IMAGE = "postgres";
     public static final Integer POSTGRESQL_PORT = 5432;
     private String databaseName = "test";
     private String username = "test";
@@ -24,6 +23,10 @@ public class PostgreSQLContainer<SELF extends PostgreSQLContainer<SELF>> extends
 
     public PostgreSQLContainer(final String dockerImageName) {
         super(dockerImageName);
+        this.waitStrategy = new LogMessageWaitStrategy()
+                .withRegEx(".*database system is ready to accept connections.*\\s")
+                .withTimes(2)
+                .withStartupTimeout(Duration.of(60, SECONDS));
     }
 
     @Override
@@ -83,20 +86,6 @@ public class PostgreSQLContainer<SELF extends PostgreSQLContainer<SELF>> extends
 
     @Override
     protected void waitUntilContainerStarted() {
-        String regEx = ".*database system is ready to accept connections.*\\s";
-        logger().info("Waiting for database to log '{}'", regEx);
-
-        WaitingConsumer waitingConsumer = new WaitingConsumer();
-        this.followOutput(waitingConsumer);
-
-
-        Predicate<OutputFrame> waitPredicate = outputFrame ->
-                outputFrame.getUtf8String().matches(regEx);
-
-        try {
-            waitingConsumer.waitUntil(waitPredicate, 120, TimeUnit.SECONDS, 2);
-        } catch (TimeoutException e) {
-            throw new ContainerLaunchException("Timed out waiting for log output matching '" + regEx + "'");
-        }
+        getWaitStrategy().waitUntilReady(this);
     }
 }

